### PR TITLE
Fix: use I3SOCK environment variable if available

### PR DESCRIPTION
**Pull request by b0o**
This fix enables i3ipc to work with i3-compatible programs such as [Sway](/swaywm/sway)

### CODE REVIEW

**Comment by Armael** [comment]
Thanks! Does that mean that after this patch the library also works with sway?

File: src/i3ipc.ml [hunk 1]
```diff
@@ -426,7 +426,10 @@ type connection = {
 
 let connect () =
   try%lwt
-    let%lwt socketpath = Lwt_process.pread ("i3", [|"i3"; "--get-socketpath"|]) in
+    let%lwt socketpath = match Sys.getenv_opt "I3SOCK" with
```
Comment (Armael) [inline src/i3ipc.ml]: Mmh, `Sys.getenv_opt` was introduced in ocaml 4.05, but at the moment the library supports ocaml >= 4.03.
Here you can quite easily rewrite the code to use `Sys.getenv` and the match-with-exception feature:
`match Sys.getenv "I3SOCK" with exception Not_found -> ... | s -> ...`

**Comment by b0o** [comment]
> Does that mean that after this patch the library also works with sway?

Yes, it works quite well with sway! So far, the only thing which doesn't seem to work (judging based on testing with the i3_msg.ml example) is the `get_bar_config` message, which seems to just hang indefinitely. I'll look into it a bit more when I have time.

I plan to be writing a few scripts to integrate with sway using this library - I will do some more testing and see if I can fix any bugs I come across.

File: src/i3ipc.ml (hunk 1 above)
Comment (b0o) [reply src/i3ipc.ml]: Thanks for the review! I'll fix this

**Comment by Armael** [comment]
>Yes, it works quite well with sway!

Very cool! Thanks for looking into this.

**Pull request merged**